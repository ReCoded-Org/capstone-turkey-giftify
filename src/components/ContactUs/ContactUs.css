.container {
  background-color: #0077b6;
  padding: 10px;
}

.container h1 {
  color: white;
  text-align: center;
  margin-bottom: 20px;
}

.contactInfo {
  color: white;
  padding: 20px;
  height: 400px;
  width: 350px;
}

.contactInfo a {
  color: white;
}

.inputForm {
  background-color: #caf0f8;
  width: 400px;
  height: 400px;
  display: flex;
  flex-direction: column;
  border-radius: 10px;
  padding: 20px;
  justify-content: space-around;
  margin: 20px;
}

.inputForm h5 {
  text-align: center;
}

.inputForm label input {
  width: 100%;
  border-color: #006eff;
  border-radius: 10px;
  background-color: white;
  color: black;
  border-width: 1px;
}

.submitData {
  align-self: center;
  width: 100px;
  background-color: #023e8a;
  color: white;
  border-radius: 20px;
}

.message {
  height: 100px;
  width: 100%;
  border-color: #006eff;
  border-radius: 10px;
  background-color: white;
  color: black;
}

.icon {
  color: #0077b6;
  background-color: rgba(255, 255, 255, 0.568);
  padding: 2px;
  border-color: #0077b6;
  border-radius: 5px;
}

.instagram {
  color: #d46aa6;
}

.facebook {
  color: #0050f1;
}

@media screen and (min-width: 800px) {
  .box {
    display: flex;
    flex-direction: row;
    justify-content: center;
  }

  .contactInfo {
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    margin: 20px;
  }

  .sosyalMed {
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    height: 100px;
  }
}

@media only screen and (max-width: 600px) {
  .box {
    display: flex;
    flex-direction: row;
    justify-content: center;
    flex-wrap: wrap;
  }

  .contactInfo {
    display: flex;
    flex-direction: column;
    justify-content: space-around;
    text-align: start;
    width: 250px;
    height: 300px;
  }

  .sosyalMed {
    display: flex;
    flex-direction: row;
    justify-content: space-between;
  }

  .inputForm {
    width: 350px;
    height: 350px;
  }
}

@media (min-width: 600px) and (max-width: 800px) {
  .box {
    display: flex;
    flex-direction: row;
    justify-content: center;
    flex-wrap: wrap;
  }

  .contactInfo {
    display: flex;
    flex-direction: column;
    justify-content: space-around;
    text-align: start;
    width: 250px;
    height: 300px;
  }

  .sosyalMed {
    display: flex;
    flex-direction: row;
    justify-content: space-between;
  }

  .inputForm {
    width: 350px;
    height: 350px;
  }
}
