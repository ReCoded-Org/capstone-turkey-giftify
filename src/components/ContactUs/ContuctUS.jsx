import React , {useState}from "react";
import { FiMapPin , FiPhone, FiMail , FiInstagram} from 'react-icons/fi'

import { AiOutlineFacebook } from 'react-icons/ai'

function ConactUS() {

    const [inputName, setInputName] = useState("");
    const [inputEmail, setInputEmail] = useState("");
    const [inputMassege, setInputMassege] = useState("");

  return (
    <div>
        <div>
            
       
            <h4> <FiMapPin /> Turkey - Istanbul</h4>
            
            <p><FiPhone /> +90 500 000 00 00</p>
            
            <p><FiMail /> example@gmail.com</p>
            
            <a href = "#"><AiOutlineFacebook /> Giftify</a>
            <br/>
           
           
            <a href = "#"> <FiInstagram /> Giftify</a>
            
        </div>

        <form>
            <label>
            Name
            <input type="text" name="name"  onChange={(e) => setInputName(e.target.value)}
 />
            Email   
            <input type="text" name="email" onChange={(e) => setInputEmail(e.target.value)} />
            Massege
            <input type="text" name="massege" onChange={(e) => setInputMassege(e.target.value)} />
            </label>
            <input type="submit" value="Submit" />
        </form>
    </div>
  );
}

export default ConactUS;
