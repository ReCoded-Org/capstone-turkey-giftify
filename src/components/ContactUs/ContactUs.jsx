import React, { useState, useEffect } from "react";
import { FiMapPin, FiPhone, FiMail } from "react-icons/fi";
import { ImFacebook2 } from "react-icons/im";
import { GrInstagram } from "react-icons/gr";
import db from "../../firebase";
import "./ContactUs.css";

function ContactUs() {
  const [inputdata, setInputData] = useState([]);
  const [inputName, setInputName] = useState("");
  const [inputEmail, setInputEmail] = useState("");
  const [inputMassege, setInputMassege] = useState("");

  function handleSubmit(e) {
    e.preventDefault();
    console.log(inputName);
    db.collection("contactUs")
      .add({
        name: inputName,
        email: inputEmail,
        massege: inputMassege,
      })
      .then(() => {
        setInputName("");
        setInputEmail("");
        setInputMassege("");
      });
  }

  useEffect(() => {
    const getData = () => {
      db.collection("contactUs").onSnapshot(function (querySnapshot) {
        setInputData(
          querySnapshot.docs.map((doc) => ({
            ...doc.data(),
            name: doc.data().inputName,
            email: doc.data().inputEmail,
            massege: doc.data().inputMassege,
          }))
        );
      });
    };
    getData();
  }, [inputdata]);

  return (
    <div className="container">
      <h1>Contact Us</h1>
      <div className="box">
        <div className="contatctinfo">
          <h5>
            {" "}
            <FiMapPin className="icon" /> Turkey - Istanbul
          </h5>
          <h5>
            <FiPhone className="icon" /> +90 500 000 00 00
          </h5>
          <h5>
            <FiMail className="icon" /> giftify@giftify.com
          </h5>
          <div className="sosyalMed">
            <h5>
              <a href="#link">
                <ImFacebook2 className="facebook" /> Giftify
              </a>
            </h5>
            <h5>
              <a href="#link">
                {" "}
                <GrInstagram className="instagram" /> Giftify
              </a>{" "}
            </h5>
          </div>
        </div>
        <div>
          <form className="inpurForm" onSubmit={handleSubmit}>
            <h5>Get In Touch</h5>
            <label>
              <h6>Name</h6>
              <input
                type="text"
                name="name"
                value={inputName}
                onChange={(e) => setInputName(e.target.value)}
              />{" "}
              <br />
              <h6>Email</h6>
              <input
                type="text"
                name="email"
                value={inputEmail}
                onChange={(e) => setInputEmail(e.target.value)}
              />{" "}
              <br />
              <h6>Massege</h6>
              <input
                className="massege"
                type="text"
                name="massege"
                value={inputMassege}
                onChange={(e) => setInputMassege(e.target.value)}
              />
            </label>
            <input className="submitData" type="submit" value="Submit" />
          </form>
        </div>
      </div>
    </div>
  );
}

export default ContactUs;
