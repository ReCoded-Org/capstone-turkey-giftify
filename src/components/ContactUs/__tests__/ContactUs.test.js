import React from "react";
import ContactUs from "./../ContactUs";
import { cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";

afterEach(cleanup);

it("matches Received", () => {
  const tree = renderer.create(<ContactUs />).toJSON();
  expect(tree).toMatchSnapshot();
});
