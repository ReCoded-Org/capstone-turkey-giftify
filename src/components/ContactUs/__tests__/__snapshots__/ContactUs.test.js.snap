// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactUs renders correctly  1`] = `
<div
  className="container"
>
  <h1>
    Contact Us
  </h1>
  <div
    className="box"
  >
    <div
      className="contactInfo"
    >
      <h5>
         
        <svg
          className="icon"
          fill="none"
          height="1em"
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M21 10c0 7-9 13-9 13s-9-6-9-13a9 9 0 0 1 18 0z"
          />
          <circle
            cx="12"
            cy="10"
            r="3"
          />
        </svg>
         Turkey - Istanbul
      </h5>
      <h5>
        <svg
          className="icon"
          fill="none"
          height="1em"
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M22 16.92v3a2 2 0 0 1-2.18 2 19.79 19.79 0 0 1-8.63-3.07 19.5 19.5 0 0 1-6-6 19.79 19.79 0 0 1-3.07-8.67A2 2 0 0 1 4.11 2h3a2 2 0 0 1 2 1.72 12.84 12.84 0 0 0 .7 2.81 2 2 0 0 1-.45 2.11L8.09 9.91a16 16 0 0 0 6 6l1.27-1.27a2 2 0 0 1 2.11-.45 12.84 12.84 0 0 0 2.81.7A2 2 0 0 1 22 16.92z"
          />
        </svg>
         +90 500 000 00 00
      </h5>
      <h5>
        <svg
          className="icon"
          fill="none"
          height="1em"
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M4 4h16c1.1 0 2 .9 2 2v12c0 1.1-.9 2-2 2H4c-1.1 0-2-.9-2-2V6c0-1.1.9-2 2-2z"
          />
          <polyline
            points="22,6 12,13 2,6"
          />
        </svg>
         giftify@giftify.com
      </h5>
      <div
        className="sosyalMed"
      >
        <h5>
          <a
            href="#link"
          >
            <svg
              className="facebook"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              version="1.1"
              viewBox="0 0 16 16"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M14.5 0h-13c-0.825 0-1.5 0.675-1.5 1.5v13c0 0.825 0.675 1.5 1.5 1.5h6.5v-7h-2v-2h2v-1c0-1.653 1.347-3 3-3h2v2h-2c-0.55 0-1 0.45-1 1v1h3l-0.5 2h-2.5v7h4.5c0.825 0 1.5-0.675 1.5-1.5v-13c0-0.825-0.675-1.5-1.5-1.5z"
              />
            </svg>
             Giftify
          </a>
        </h5>
        <h5>
          <a
            href="#link"
          >
             
            <svg
              className="instagram"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M17.3183118,0.0772036939 C18.5358869,0.132773211 19.3775594,0.311686093 20.156489,0.614412318 C20.9357539,0.917263935 21.5259307,1.30117806 22.1124276,1.88767349 C22.6988355,2.47414659 23.0827129,3.06422396 23.3856819,3.84361655 C23.688357,4.62263666 23.8672302,5.46418415 23.9227984,6.68172489 C23.9916356,8.19170553 24,8.72394829 24,11.9999742 C24,15.2760524 23.9916355,15.808302 23.9227954,17.3182896 C23.8672306,18.5358038 23.6883589,19.3773584 23.3855877,20.1566258 C23.0826716,20.9358162 22.6987642,21.5259396 22.1124276,22.1122749 C21.5259871,22.6987804 20.9357958,23.0827198 20.1563742,23.3856323 C19.3772192,23.6883583 18.5357324,23.8672318 17.3183209,23.9227442 C15.8086874,23.9916325 15.2765626,24 12,24 C8.72343739,24 8.19131258,23.9916325 6.68172382,23.9227463 C5.46426077,23.8672314 4.62270711,23.6883498 3.84342369,23.3855738 C3.0641689,23.0827004 2.47399369,22.6987612 1.88762592,22.1123283 C1.30117312,21.525877 0.91721975,20.9357071 0.614318116,20.1563835 C0.311643016,19.3773633 0.132769821,18.5358159 0.0772038909,17.3183251 C0.0083529426,15.8092887 0,15.2774634 0,11.9999742 C0,8.7225328 0.00835296697,8.19071076 0.0772047368,6.68165632 C0.132769821,5.46418415 0.311643016,4.62263666 0.614362729,3.84350174 C0.91719061,3.06430165 1.30113536,2.4741608 1.88757245,1.88772514 C2.47399369,1.30123879 3.0641689,0.917299613 3.84345255,0.614414972 C4.62236201,0.311696581 5.46409415,0.132773979 6.68163888,0.0772035898 C8.19074867,0.00835221992 8.72252573,0 12,0 C15.2774788,0 15.8092594,0.00835235053 17.3183118,0.0772036939 Z M12,2.66666667 C8.75959504,2.66666667 8.26400713,2.67445049 6.80319929,2.74109814 C5.87614637,2.78341009 5.31952221,2.90172878 4.80947575,3.09995521 C4.37397765,3.26922052 4.09725505,3.44924273 3.77324172,3.77329203 C3.44916209,4.09737087 3.26913181,4.37408574 3.09996253,4.80937168 C2.90169965,5.31965737 2.78340891,5.87618164 2.74109927,6.80321713 C2.67445122,8.26397158 2.66666667,8.75960374 2.66666667,11.9999742 C2.66666667,15.2403924 2.67445121,15.7360281 2.74109842,17.1967643 C2.78340891,18.1238184 2.90169965,18.6803426 3.09990404,19.1904778 C3.26914133,19.6259017 3.44919889,19.9026659 3.77329519,20.2267614 C4.09725505,20.5507573 4.37397765,20.7307795 4.80932525,20.8999863 C5.31971515,21.0982887 5.87621193,21.2165784 6.80323907,21.2588497 C8.26460439,21.3255353 8.76051223,21.3333333 12,21.3333333 C15.2394878,21.3333333 15.7353956,21.3255353 17.1968056,21.2588476 C18.123775,21.216579 18.6802056,21.0982995 19.1905083,20.9000309 C19.6260288,20.7307713 19.9027426,20.5507596 20.2267583,20.226708 C20.5507492,19.9027179 20.7308046,19.6259456 20.9000375,19.1906283 C21.0983009,18.6803412 21.2165908,18.1238118 21.2588986,17.196779 C21.3255376,15.7350718 21.3333333,15.2390126 21.3333333,11.9999742 C21.3333333,8.76098665 21.3255376,8.26493375 21.2589016,6.80323567 C21.2165911,5.87618164 21.0983004,5.31965737 20.9001178,4.80957831 C20.7308131,4.37403932 20.550774,4.09729207 20.2267583,3.77324038 C19.9027658,3.44924868 19.6260264,3.26922777 19.1905015,3.09996643 C18.6803988,2.90171817 18.1238378,2.78341062 17.1967608,2.74109868 C15.7359966,2.67445057 15.2404012,2.66666667 12,2.66666667 Z M12,18.2222222 C8.56356156,18.2222222 5.77777778,15.4364384 5.77777778,12 C5.77777778,8.56356156 8.56356156,5.77777778 12,5.77777778 C15.4364384,5.77777778 18.2222222,8.56356156 18.2222222,12 C18.2222222,15.4364384 15.4364384,18.2222222 12,18.2222222 Z M12,15.5555556 C13.9636791,15.5555556 15.5555556,13.9636791 15.5555556,12 C15.5555556,10.0363209 13.9636791,8.44444444 12,8.44444444 C10.0363209,8.44444444 8.44444444,10.0363209 8.44444444,12 C8.44444444,13.9636791 10.0363209,15.5555556 12,15.5555556 Z M18.2222222,7.11111111 C17.4858426,7.11111111 16.8888889,6.51415744 16.8888889,5.77777778 C16.8888889,5.04139811 17.4858426,4.44444444 18.2222222,4.44444444 C18.9586019,4.44444444 19.5555556,5.04139811 19.5555556,5.77777778 C19.5555556,6.51415744 18.9586019,7.11111111 18.2222222,7.11111111 Z"
              />
            </svg>
             Giftify
          </a>
           
        </h5>
      </div>
    </div>
    <div>
      <form
        className="inputForm"
        onSubmit={[Function]}
      >
        <h5>
          Get In Touch
        </h5>
        <label>
          <h6>
            Name
          </h6>
          <input
            name="name"
            onChange={[Function]}
            type="text"
            value=""
          />
           
          <br />
          <h6>
            Email
          </h6>
          <input
            name="email"
            onChange={[Function]}
            type="text"
            value=""
          />
           
          <br />
          <h6>
            Message
          </h6>
          <textarea
            className="Message"
            name="Message"
            onChange={[Function]}
            type="Message"
            value=""
          />
        </label>
        <input
          className="submitData"
          type="submit"
          value="Submit"
        />
      </form>
    </div>
  </div>
</div>
`;
